contracts/distributions/TokenDistributor.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+/*
+      /$$$$$$            /$$$$$$$   /$$$$$$   /$$$$$$
+     /$$__  $$          | $$__  $$ /$$__  $$ /$$__  $$
+    | $$  \__/ /$$   /$$| $$  \ $$| $$  \ $$| $$  \ $$
+    |  $$$$$$ | $$  | $$| $$  | $$| $$$$$$$$| $$  | $$
+     \____  $$| $$  | $$| $$  | $$| $$__  $$| $$  | $$
+     /$$  \ $$| $$  | $$| $$  | $$| $$  | $$| $$  | $$
+    |  $$$$$$/|  $$$$$$/| $$$$$$$/| $$  | $$|  $$$$$$/
+     \______/  \______/ |_______/ |__/  |__/ \______/
+
+*/
+pragma solidity ^0.8.12;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+
+import "../access-control/SuAccessControlAuthenticated.sol";
+import "../interfaces/ITokenDistributor.sol";
+import "../interfaces/IveERC20v2.sol";
+
+/**
+ * @title The contract that distribute suDAO tokens for community based on NFT membership
+ */
+contract TokenDistributor is SuAccessControlAuthenticated, ITokenDistributor {
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    IveERC20v2 public VE_ERC_20;
+    IERC20Upgradeable public SU_DAO;
+    IERC20Upgradeable public SU_DAO_OLD;
+
+    function initialize(
+        address _accessControlSingleton,
+        address _suDAOOld,
```

Comment:
suDAOOld and _suDAONew might be written as _suDaoOld and _suDaoNew. This can make it clearer where one word ends and the next begins.